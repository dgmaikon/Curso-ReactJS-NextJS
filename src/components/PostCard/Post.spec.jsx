import { render, screen } from "@testing-library/react";
import { describe } from "vitest";
import { PostCard } from ".";
import { PostCardPropsMock } from "./mock";

const props = PostCardPropsMock;

describe("PostCard", () => {
    it("should render PostCard correctly", () => {
      render(<PostCard {...props}/>);

      expect(screen.getByAltText(/title1/))
      .toHaveAttribute('src', props.cover);
      expect(screen.getByRole('heading', {name: /title1/}))
      .toBeInTheDocument();
      expect(screen.getByText(/body1/))
      .toBeInTheDocument();
    });

    it("should macth snapshot", () => {
        const {container} = render(<PostCard {...props}/>);
        expect (container.firstChild).toMatchSnapshot();
    });
})
