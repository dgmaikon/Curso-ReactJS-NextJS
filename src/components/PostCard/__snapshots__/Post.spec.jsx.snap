// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PostCard > should macth snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="title1"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h2>
      title1
       
      1
    </h2>
    <p>
      body1
    </p>
  </div>
</div>
`;
